core/src/main/java/io/confluent/rest/metrics/MetricsResourceMethodApplicationListener.java

```diff
@@ -256,7 +319,24 @@ public void finished(long requestSize, long responseSize, long latencyMs) {
     /**
      * Indicate that a request has failed with an exception.
      */
-    public void exception() {
+    public void exception(final RequestEvent event) {
+      int errorCode = -1;
+      if (event.getException() instanceof RestException) {
```

Comment:
If you're looking for classifying http status codes, `RestException` and `getErrorCode()` aren't quite right. `RestException` inherits from the more general `WebApplicationException`. The error code is an extension that `rest-utils` puts in place to allow more granularity and standardize the output format for the body of errors to include the more detailed version. e.g. if schema registry doesn't find a subject, the status is 404, but the error code is 40401 so you can classify the issue more specifically (in an automated way, without relying on the text of the error message).

I don't know the details of the motivation here, but you probably want to use `WebApplicationException` and `getStatus()`. If not, you have to deal with more of range of errors since most exceptions thrown by actual rest-utils applications use error codes with 5 digits (3 for http status, 2 additional to classify with more granularity), so only things like framework-provided errors would be covered by this (which works fine for everything other than 4-500 ranges, but presumably those are the ones you're interested in granular metrics for generally).

Reply:
Why not look at the Response status which has the https status rather than evaluating the Exception?

Reply:
Sure, that works too. Key point being that we want status, not error code.

Reply:
I think the best strategy here is to use `WebApplicationException..getResponse().getStatus()`. It is as general as we can get and still pretty clean. Sound good?